include/openmc/material.h

```diff
@@ -33,6 +33,21 @@ extern std::unordered_map<int32_t, int32_t> material_map;
 //! A substance with constituent nuclides and thermal scattering data
 //==============================================================================
 
+struct PolyProperty{
+  std::string type_;                       //! user readable type
+  double coeffs_[100];                     //! coefficients for poly evaluation
+  int n_coeffs_;                           //! number of coeffs
+  int order_;                              //! the order of the expansion
+  int geom_norm_offset_;                   //! offset for the geom_norms coeffs
+  double poly_results_[100];               //! variables used in evaluation property
+  double poly_norm_[100];                  //! polynomial norm available to property for efficiency
```

Comment:
All these statically-sized arrays should be `std::vector<double>` or `xt::xtensor<double, 1>`.